systest/21million/test-21million.sh

```diff
@@ -80,17 +80,6 @@ if [[ -n $SAVEDIR ]]; then
     QUIET=yes
 fi
 
-# check data version to help distinguish diffs due to a change in the data
```

Comment:
Yup, this ETAG check is being removed since the data set is no longer being downloaded every single time, so this step is no longer needed. Checking if the data changed across test runs seems like a good idea though, so it can be brought back in a later PR if data changes end up causing test issues.